extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/SplitPackageProcessor.java

```diff
@@ -107,4 +138,30 @@ void splitPackageDetection(ApplicationArchivesBuildItem archivesBuildItem,
                     "Following packages were detected in multiple archives: %s", splitPackagesWarning.toString());
         }
     }
+
+    private List<Predicate<String>> initPredicates(Collection<String> exclusions) {
+        final String packMatch = ".*";
+        List<Predicate<String>> predicates = new ArrayList<>();
+        for (String exclusionExpression : exclusions) {
+            if (exclusionExpression.endsWith(packMatch)) {
+                // Package ends with ".*"
+                final String pack = exclusionExpression.substring(0, exclusionExpression.length() - packMatch.length());
```

Comment:
Wondering if we should make things a bit more complex. E.g. either the package should be `my.package` or it should start with `my.package.` (with a final dot). I don't think we want to catch `my.package2` and if I'm not mistaken, we catch it with what you did.

Reply:
This comment should into the original PR :)

Reply:
Current way actually allows for both cases. If you really wanted to, you could filter with `my.package..*` (note the double dot) to get the behavior.
But yes, with just `my.package.*`, `my.package2` will be considered a match.
Honestly, I think we are going into enough complexity already for a configuration which has the purpose of suppressing a warning which you should have fixed and not suppressed in the first place :-)

Reply:
Well, I disagree. The current behavior is incorrect, it needs a fix. And I don't think it will be that hard. If you're not willing to do it, I can do it.